app/controllers/posts_controller.rb

```diff
@@ -11,5 +12,21 @@ def show
     @user = User.find(user_id)
     @post = Post.find(post_id)
     @comments = Comment.where(post_id)
+    @recent_user = current_user
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    user = current_user
+    @post = Post.new(
+      title: params[:title],
+      text: params[:text],
+      author_id: user.id
+    )
+    return unless @post.save
+    redirect_to user_posts_path
```

Comment:
- [OPTIONAL] I would advise you to use **_strong params_** instead of accessing the params directly. Strong Params allow developers to specify in the controller which parameters are accepted and used. By permitting only the expected params, any unneeded or potentially dangerous params will be ignored and effectively filtered out. You can get more information on strong params by clicking the link provided in this [article](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides.) 😎